config/interpolate_funcs.go

```diff
@@ -1548,6 +1549,50 @@ func interpolationFuncURLEncode() ast.Function {
 	}
 }
 
+// interpolationFuncTranspose implements the "transpose" function
+// that converts a map (string,list) to a map (string,list) where
+// the unique values of the original lists become the keys of the
+// new map and the keys of the original map become values for the
+// corresponding new keys.
+func interpolationFuncTranspose() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeMap},
+		ReturnType: ast.TypeMap,
+		Callback: func(args []interface{}) (interface{}, error) {
+
+			inputMap := args[0].(map[string]ast.Variable)
+			outputMap := make(map[string]ast.Variable)
+			tmpMap := make(map[string][]string)
+
+			for inKey, inVal := range inputMap {
+				values := inVal.Value.([]ast.Variable)
```

Comment:
I think `inVal.Value` isn't necessarily a slice here... the signature of this function just guarantees that it's a map, but doesn't say anything about the map's element type.

I think this could be addressed by checking `inVal.Type` here to ensure that it's `ast.TypeList`, producing an error if not:

```go
if inVal.Type != ast.TypeList {
    return nil, fmt.Errorf("transpose requires a map of lists of strings")
}
```

Reply:
I agree. A more explicit check prior to the type assertion with early return seems appropriate.